databases/backends/postgres.py

```diff
@@ -74,18 +74,21 @@ def __init__(self, row: tuple, result_columns: tuple, dialect: Dialect) -> None:
             for idx, (column_name, _, _, datatype) in enumerate(self._result_columns)
         }
 
-    def __getitem__(self, key: str) -> typing.Any:
-        idx, datatype = self._column_map[key]
-        raw = self._row[idx]
+    def __getitem__(self, key: typing.Any) -> typing.Any:
```

Comment:
Not happy with this implementation in general and especially passing `typing.Any`. I think it might be limited to `str` and `int` as a raw record's fields can be accessed by an int index or by the str key.